rules_go tests failing with Bazel@HEAD

First reported at https://github.com/bazelbuild/continuous-integration/issues/1429
https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/2632#01833a2a-4026-47c4-9730-89252449e823
```
    --- FAIL: TestLcovCoverage/without-race (5.06s)
        lcov_coverage_test.go:171: bazel-testlogs/src/lib_test/coverage.dat: does not contain:
            
            SF:src/other_lib.go
            FNF:0
            FNH:0
            DA:3,1
            DA:4,1
            DA:5,0
            DA:6,0
            DA:7,1
            LH:3
            LF:5
            end_of_record
            
            actual content:
            
```

`//tests/core/coverage:lcov_coverage_test` and `//tests/core/coverage:lcov_test_main_coverage_test` are broken by coverage_output_generator v2.6 introduced at https://github.com/bazelbuild/bazel/commit/a70d84b9eaae99f665ccdebd68eb3da57e07530b

(There is another test `//tests/core/cross:non_executable_test` also failing, but due a Bazel issue, we are investigating)

--- a/go/private/rules/library.bzl
+++ b/go/private/rules/library.bzl
@@ -52,6 +52,12 @@ def _go_library_impl(ctx):
         DefaultInfo(
             files = depset([archive.data.file]),
         ),
+        coverage_common.instrumented_files_info(
+            ctx,
+            source_attributes = ["srcs"],
+            dependency_attributes = ["data", "deps", "embed", "embedsrcs"],
+            extensions = ["go"],
+        ),
         OutputGroupInfo(
             cgo_exports = archive.cgo_exports,
             compilation_outputs = [archive.data.file],

--- a/go/private/rules/test.bzl
+++ b/go/private/rules/test.bzl
@@ -181,7 +181,7 @@ def _go_test_impl(ctx):
         coverage_common.instrumented_files_info(
             ctx,
             source_attributes = ["srcs"],
-            dependency_attributes = ["deps", "embed"],
+            dependency_attributes = ["data", "deps", "embed", "embedsrcs"],
             extensions = ["go"],
         ),
         testing.TestEnvironment(env),

